pkg/virt-controller/watch/job.go

```diff
@@ -55,12 +50,9 @@ type JobDispatch struct {
 
 func (jd *JobDispatch) Execute(store cache.Store, queue workqueue.RateLimitingInterface, key interface{}) {
 	// Fetch the latest Job state from cache
-	obj, exists, err := store.GetByKey(key.(string))
+	// Cannot return an error
+	obj, exists, _ := store.GetByKey(key.(string))
```

Comment:
The controller loop has no idea which cache implementation is used, so I would say that it is not it's job to guess about the internals.